### PR TITLE
Sonar integration

### PR DESCRIPTION
**Pull request by rgildein**
- update newer templates
- add sonar

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,12 +76,12 @@ unittests:
 	@echo "Running unit tests"
 	@tox -e unit -- ${UNIT_ARGS}
 
-functional:
-	@echo "Executing functional tests(charm will be build during functional tests)"
-	@tox -e func -- ${FUNC_ARGS}
+functional:
```
Comment (agileshaw) [inline Makefile]: It seems that a `build` target is needed here.
```suggestion
functional: build
```

**Review by Pjack — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (rgildein) [reply Makefile]: No it's not. The pytest-operator will handle charm build for us.

Comment (agileshaw) [reply Makefile]: Noted. Then could you please remove the trailing blank space at the end?
```suggestion
functional:
```

**Review by agileshaw — CHANGES_REQUESTED** [review]

**Review by agileshaw — APPROVED** [review]

**Pull request merged**